buildSrc/src/main/java/org/elasticsearch/gradle/JdkDownloadPlugin.java

```diff
@@ -111,83 +119,162 @@ private static void setupRootJdkDownload(Project rootProject, String platform, S
         String hash = jdkVersionMatcher.group(5);
 
         // add fake ivy repo for jdk url
-        String repoName = REPO_NAME_PREFIX + version;
+        String repoName = REPO_NAME_PREFIX + vendor + "_" + version;
         RepositoryHandler repositories = rootProject.getRepositories();
         if (rootProject.getRepositories().findByName(repoName) == null) {
-            if (hash != null) {
-                // current pattern since 12.0.1
+            if (vendor.equals("adoptopenjdk")) {
+                if (hash != null) {
+                    throw new IllegalArgumentException("adoptopenjdk versions do not have hashes but was [" + version + "]");
+                }
                 repositories.ivy(ivyRepo -> {
                     ivyRepo.setName(repoName);
-                    ivyRepo.setUrl("https://download.oracle.com");
+                    ivyRepo.setUrl("https://artifactory.elstc.co/");
```

Comment:
The repo URL here should be `https://artifactory.elstc.co/artifactory/oss-jdk-local` since that's the actual base URL of the repository. I'm not sure if using the Artifactory server URL here will cause funkiness with artifact caching since Gradle might thing all repos with that URL are the same repo, when in fact, they are not.